commcare_connect/program/api/serializers.py

```
@@ -0,0 +1,330 @@
+        return data
+
+    def _resolve_api_key(self, user, hq_server, api_key_string):
+        try:
+            api_key, __ = HQApiKey.objects.get_or_create(
+                user=user,
+                hq_server=hq_server,
+                api_key=api_key_string,
+            )
+            return api_key
+        except IntegrityError:
+            raise serializers.ValidationError({"api_key": _("This API key is already registered to another user.")})
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)